parachain/integration-test/src/lib.rs

```diff
@@ -0,0 +1,391 @@
+#![cfg(test)]
+
+use anyhow::anyhow;
+use hex::ToHex;
+use ismp::{
+	consensus::{StateMachineHeight, StateMachineId},
+	events::{Event, StateMachineUpdated},
+	host::{StateMachine, StateMachine::Kusama},
+};
+use log::log;
+use sc_service::TaskManager;
+use sp_keyring::sr25519::Keyring;
+use std::{
+	collections::{BTreeMap, HashMap},
+	ops::Div,
+	sync::Arc,
+	time::Duration,
+};
+use subxt::{
+	ext::{
+		codec::Encode,
+		sp_core::{sr25519::Pair, Pair as PairT},
+	},
+	rpc_params,
+	storage::StorageAddress,
+	tx::{PairSigner, Signer},
+	utils::AccountId32,
+	OnlineClient,
+};
+use subxt_signer::sr25519::{
+	dev::{self},
+	Keypair,
+};
+use subxt_utils::{
+	gargantua::{
+		api,
+		api::{
+			runtime_types,
+			runtime_types::{
+				gargantua_runtime::RuntimeCall,
+				pallet_ismp_demo::pallet::{GetRequest, TransferParams},
+			},
+		},
+	},
+	Hyperbridge,
+};
+use tesseract::logging::setup as log_setup;
+use tesseract_config::AnyConfig;
+use tesseract_messaging::{events::translate_events_to_messages, handle_notification, relay};
+use tesseract_primitives::{config::RelayerConfig, IsmpProvider, TxReceipt};
+use tesseract_substrate::{config::KeccakSubstrateChain, SubstrateClient, SubstrateConfig};
+use toml::Table;
+use tracing_subscriber::{filter::LevelFilter, util::SubscriberInitExt};
+use subxt_utils::gargantua::api::ismp::events::Request;
+use subxt_utils::gargantua::api::ismp_demo::events::BalanceTransferred;
+use transaction_fees::TransactionPayment;
+
+// This was the effort to mitigate the issue of the current relayer as it assumes there is a
+// coprocessor in place
+async fn messaging_relayer_lite(
```

Comment:
You should not do this, we want to ensure we are testing the actual code that would be used in production, I don't think there's any issue using the current messaging relay, Ignore the coprocessor, it would still work if you set a random coprocessor.

There might be errors with fee accumulation but you can ignore those.